src/device-config.coffee

```diff
@@ -89,28 +70,30 @@ module.exports = class DeviceConfig
 		}
 		db('deviceConfig').update(confToUpdate)
 
-	filterConfigKeys: (conf) =>
-		_.pickBy conf, (v, k) =>
-			_.includes(@validKeys, k) or _.startsWith(k, bootConfigEnvVarPrefix)
-
 	getTarget: ({ initial = false } = {}) =>
 		@db.models('deviceConfig').select('targetValues')
-		.then ([ devConfig ]) ->
-			return JSON.parse(devConfig.targetValues)
-		.then (conf) =>
-			conf = @filterConfigKeys(conf)
+		.then ([ devConfig ]) =>
+			return Promise.all [
```

Comment:
`Promise.join` :)

Reply:
I prefer to stick with `Promise.all` when I don't need to worry about performance, mainly because you can keep the same indentation level, and they're equivalent (again ignoring performance).